### PR TITLE
BUG: Fix bug with nan, update infrastructure

### LINKED ISSUE #289: MAINT: Test with mne-qt-browser
The default plotting backend for MNE is mne-qt-browser, but it's untested here. For me this example fails:

https://autoreject.github.io/stable/auto_examples/plot_visualize_bad_epochs.html#sphx-glr-auto-examples-plot-auto-repair-py

I think we need:
```diff
diff --git a/autoreject/autoreject.py b/autoreject/autoreject.py
index b0fdb09..2b03da9 100644
--- a/autoreject/autoreject.py
+++ b/autoreject/autoreject.py
@@ -1422,7 +1422,7 @@ class RejectLog(object):
                 if not np.isnan(this_label):
                     epoch_color.append(color_map[this_label])
                 else:
-                    epoch_color.append(None)
+                    epoch_color.append('k')
             epoch_colors.append(epoch_color)
 
         return plot_mne_epochs(
diff --git a/examples/plot_visualize_bad_epochs.py b/examples/plot_visualize_bad_epochs.py
index a42757e..f372e0e 100644
--- a/examples/plot_visualize_bad_epochs.py
+++ b/examples/plot_visualize_bad_epochs.py
```
But this should be tested as part of CI integration of mne-qt-browser as well.

Ideally the mne-qt-browser scraper from MNE would be used to build docs as well.

### PR DESCRIPTION
**Pull request by larsoner**
1. Update to use MNE-Python `browser_backend` fixture. This comes with some nice extra benefits, like treating warnings as errors but inheriting MNE-Python `main`'s latest `ignores` list.
2. Update tests with `verbose='error'` and `ignorewarnings` to deal with warnings that were silently present before
3. Update to use `testing` data for all tests, rather than `sample`, `fnirs_motor`, and `misc`
4. Download and cache the `testing` dataset in tests on Azure and GH-actions using MNE-Python infrastructure
5. Add `mne-qt-browser` backend testing to GH actions
6. Fix bug with `colors.append(None)` which actually caused errors on *both* mpl and qt backends, but wasn't being hit because it was not on a covered line (it is now)
7. Remove conditional on codecov reporting since `codecov` should be smart enough nowadays to combine all reports
8. Add concurrency auto-cancellation of previous GH actions runs when pushing new commits
9. Update 3.8->3.10 so that oldest (3.7) and newest (3.10) are tested

Closes #289

### CODE REVIEW

**Comment by larsoner** [comment]
Okay I ended up copying parts of the MNE-Python `conftest.py` because the `pytest_plugins` mechanism refused to work. Seems worth the minor code dup for the benefits we get. Should come back green this time -- ready for review/merge from my end

**Comment by larsoner** [comment]
(branch protection rules will need to be changed to reflect the updated names, which I can do if people are otherwise happy with this PR)

File: .github/workflows/test.yml
Comment (sappelhoff) [inline .github/workflows/test.yml:113]: why specify the branch twice?

```suggestion
      run: git clone --depth 1 --single-branch --branch main https://github.com/mne-tools/mne-python.git
```

**Review by sappelhoff — APPROVED** [review]
lots of magic in the conftest fail, but I agree with the changes overall (didn't dive into details)

**Comment by larsoner** [comment]
> lots of magic in the conftest file

Indeed but it's all copy-paste-adapted from MNE-Python so I think we should be safe here, and updates hopefully won't be too painful to make (since we can just look at MNE-Python to see what needs changing/updating)

**Comment by sappelhoff** [comment]
@larsoner could you adjust the required workflows (branch protection)? After that, I think we can hit auto-merge, because this is infra-only, so I don't think we desperately need another review.

File: autoreject/utils.py
Comment (jasmainak) [inline autoreject/utils.py:450]: `'error'` or `verbose` ?

**Comment by jasmainak** [comment]
Looks good @larsoner ! Would try to make doubly sure before merging that changing the testing dataset doesn't change the test somehow ... thanks for all the fixes! I can't keep up with changes in the MNE main :)

File: autoreject/utils.py
Comment (larsoner) [reply autoreject/utils.py:456]: 'error' is fine here -- the only arg to `use_log_level` is `verbose` so it's implicitly `verbose='error'` (but in practice always written as `use_log_level('info')` etc. in MNE)

**Pull request merged**
